### PR TITLE
cleaning up the readme and moving docs into readthedocs

### PR DESCRIPTION
**Pull request by chayim**

### CODE REVIEW

File: docs/advanced_features.rst [hunk 1]
```diff
@@ -0,0 +1,436 @@
...
+   ...             # we just did happened atomically.
+   ...             break
+   ...        except WatchError:
+   ...             # another client must have changed 'OUR-SEQUENCE-KEY' between
+   ...             # the time we started WATCHing it and the pipeline's execution.
+   ...             # our best bet is to just retry.
+   ...             continue
+
+Note that, because the Pipeline must bind to a single connection for the
+duration of a WATCH, care must be taken to ensure that the connection is
```
Comment (nermiller) [inline docs/advanced_features.rst:114]:
```suggestion
duration of a WATCH, you need to ensure that the connection is
```

File: docs/advanced_features.rst [hunk 2]
```diff
@@ -0,0 +1,436 @@
...
+returned to the connection pool by calling the reset() method. If the
+Pipeline is used as a context manager (as in the example above) reset()
+will be called automatically. Of course you can do this the manual way
```
Comment (nermiller) [inline docs/advanced_features.rst:117]:
```suggestion
is called automatically. You can do this manually
```

File: docs/advanced_features.rst [hunk 3]
```diff
@@ -0,0 +1,436 @@
...
+   ...         pipe.watch('OUR-SEQUENCE-KEY')
+   ...         ...
+   ...         pipe.execute()
+   ...         break
+   ...     except WatchError:
+   ...         continue
+   ...     finally:
+   ...         pipe.reset()
+
+A convenience method named "transaction" exists for handling all the
```
Comment (nermiller) [inline docs/advanced_features.rst:134]:
```suggestion
A convenience method named "transaction" is used 
```

File: docs/advanced_features.rst [hunk 4]
```diff
@@ -0,0 +1,436 @@
...
+boilerplate of handling and retrying watch errors. It takes a callable
```
Comment (nermiller) [inline docs/advanced_features.rst:135]:
```suggestion
to handle and retry watch errors. It takes a callable
```

File: docs/advanced_features.rst [hunk 5]
```diff
@@ -0,0 +1,436 @@
...
+that should expect a single parameter, a pipeline object, and any number
+of keys to be WATCHed. Our client-side INCR command above can be written
```
Comment (nermiller) [inline docs/advanced_features.rst:137]:
```suggestion
of keys to be WATCHed. You can write the client-side INCR command above 
```

File: docs/advanced_features.rst [hunk 6]
```diff
@@ -0,0 +1,436 @@
...
+   >>> def client_side_incr(pipe):
+   ...     current_value = pipe.get('OUR-SEQUENCE-KEY')
+   ...     next_value = int(current_value) + 1
+   ...     pipe.multi()
+   ...     pipe.set('OUR-SEQUENCE-KEY', next_value)
+   >>>
+   >>> r.transaction(client_side_incr, 'OUR-SEQUENCE-KEY')
+   [True]
+
+Be sure to call pipe.multi() in the callable passed to Redis.transaction
```
Comment (nermiller) [inline docs/advanced_features.rst:151]:
```suggestion
Prior to any write commands, call pipe.multi() in the callable passed to Redis.transaction.
```

File: docs/advanced_features.rst [hunk 7]
```diff
@@ -0,0 +1,436 @@
...
+prior to any write commands.
```
Comment (nermiller) [inline docs/advanced_features.rst:152]:
```suggestion

```

File: docs/advanced_features.rst [hunk 8]
```diff
@@ -0,0 +1,436 @@
+Advanced Features
```
Comment (nermiller) [inline docs/advanced_features.rst:1]:
```suggestion
Advanced features
```

**Review by nermiller — APPROVED** [review]
LGTM. Just some minor edits here.

**Review by dvora-h — APPROVED** [review]

**Pull request merged**